twister: Fix quarantine filtering simulations

gchwier opened the pull request:
Quarantine filter was not correctly identifying the simulator name. Now it uses the simulator name from the Platform object, ensuring that quarantined tests are properly excluded or verified.

Tested with:
* mps2/an521/cpu0 supports two simulators: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mps2/mps2_an521_cpu0.yaml#L7

Create `quarantine.yaml` file:
```
cat quarantine.yaml
- simulations:
  - armfvp
  comment: "armfvp"
- simulations:
  - qemu
  comment: "qemu"
```
Check if quarantine works for the default simulations (qemu):
`./scripts/twister -T samples/hello_world -p mps2/an521/cpu0 -vv -ll debug -c --quarantine-list quarantine.yaml`
logged:
`DEBUG   - mps2/an521/cpu0           samples/hello_world/sample.basic.helloworld        FILTERED: Quarantine: qemu`

Check if quarantine works with non-default simulator (armfvp):
`./scripts/twister -T samples/hello_world -p mps2/an521/cpu0 -vv -ll debug -c --simulation armfvp --quarantine-list quarantine.yaml`
logged:
`DEBUG   - mps2/an521/cpu0           samples/hello_world/sample.basic.helloworld        FILTERED: Quarantine: armfvp`

Run with other simulations:
`./scripts/twister -T samples/hello_world -p qemu_x86 -p native_sim -vv -ll debug -c --quarantine-list quarantine.yaml`
logged
```
DEBUG   - qemu_x86/atom             samples/hello_world/sample.basic.helloworld        FILTERED: Quarantine: qemu
...
INFO    - 1/1 native_sim/native         samples/hello_world/sample.basic.helloworld        PASSED (native 2.623s <host>)
```

Issue introduced by: https://github.com/zephyrproject-rtos/zephyr/pull/79174

gchwier committed:
twister: Fix quarantine filtering simulations

Quarantine filter was not correctly identifying the simulator name.
Now it uses the simulator name from the Platform object,
ensuring that quarantined tests are properly excluded or verified

Signed-off-by: Grzegorz Chwierut <grzegorz.chwierut@nordicsemi.no>

golowanow reviewed (commented):
thank you for fixing this! 
could you also add a missing example of quarantine.yaml with simulations here
https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/develop/test/twister.rst?plain=1#L1529

gchwier committed:
doc: twister: update the quarantine section

Update the quarantine section in the twister.rst documentation
to include information about excluding specific platforms,
architectures, and simulators

Signed-off-by: Grzegorz Chwierut <grzegorz.chwierut@nordicsemi.no>

gchwier commented:
> thank you for fixing this! could you also add a missing example of quarantine.yaml with simulations here https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/develop/test/twister.rst?plain=1#L1529

thanks for review, updated docs
